### PR TITLE
Make EduBot installable and add CI/CD pipeline

### PR DESCRIPTION
**Pull request by skilkis**
This code-quality pull-request first and foremost makes EduBot installable through use of a `setup.py` and `setup.cfg` file, where the latter contains all metadata and configuration options. To accommodate packaging in a clean way and make use of its testing benefits, the `src/` Python directory layout is used. This layout is becoming popular due to it's benefits of allowing developers to consistently test the installed version of the package instead of the local source code (Pros/Cons Discussed in [1]).

After the package has been made installable, a CI/(CD to be added later) pipeline using GitHub actions was defined that runs the following checks for both Python 3.7 and 3.8 in parallel.

1. Checking package Installation on both Python 3.7 and 3.8
2. Code Linting with flake8 (configured in setup.py but currently allowed to fail).
    a. Full spectrum of PEP-8 checks using pycodestyle
    b. Checks for formatting using Black
    c. Checks docstrings for compliance with Google docstring convention
    d. Checks import statement order using isort (Conforming to [2])
3. Running unit-tests with Pytest and generating code coverage report
4. Checking coverage report to ensure minimum coverage (30%) is reached

Currently, the testing of the code has proven difficult and only an initial test for the `EduBot` class has been completed. Additionally, the mocking helpers used for some reason do not work on Python 3.7 which is the reason for the failing status of the CI/CD pipeline.

**Summary of Changes**:
- Add setup.cfg and setup.py file to make `EduBot` installable using `pip`
- Transition to using the `src/` directory layout to allow testing of the installed package
- Move and rename `edubot.py` to `src/edubot/bot.py` to prevent naming conflicts
- Add AGPL-3.0 copyright notice to some source files (Pending approval @jooste)
- Transition to using `pkg_resources` for referencing static `.json` files
- `.editorconfig` file to normalize whitespace and EOL across all developers
- Add `CONTRIBUTING.md` file to help developers install `EduBot` and get informed about best practices
- Add CI/CD Pipeline using GitHub Actions
- Add discord.py mock testing capability using test helpers from [3]
- Add initial unit-test of the `EduBot` class using pytest
- Merge previous Pull-Requests 6330cd4890c9966617789af0988b85610b3906fd 74f015441d4fdc1a228fb60a5cb750433aac8f1b 3644e460b88c9b8d7d47df4a11cf3778f16d9348 308875b21ca33c5d76eee979c473311e85fc2301

@jooste  I have taken care to merge all previous pull-requests that came in after I had forked the repo, therefore this commit is ready to be merged into the `master` branch. Let me know if you have questions or would like to discuss the changes. 

[1]: https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure
[2]: https://google.github.io/styleguide/pyguide.html#313-imports-formatting
[3]: https://github.com/python-discord/bot

### CODE REVIEW

**Comment by skilkis** [comment]
@jooste I resolved merge conflicts due to #14 for you, it's ready to be merged if you approve of the changes 😃

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    online-lecturing
+license = Apache License Version 2.0
+license-file = LICENSE
+url = https://github.com/TUDelft-CNS-ATM/EduBot
+author = Joost Ellerbroek
+author_email = j.ellerbroek@tudelft.nl
+classifiers =
+    Development Status :: 3 - Alpha
+    Intended Audience :: Developers
+    License :: OSI Approved :: Apache License Version 2.0
```
Comment (jooste) [inline setup.cfg]: Shouldn't this be AGPLv3?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    python
+    online-lecturing
+license = Apache License Version 2.0
+license-file = LICENSE
+url = https://github.com/TUDelft-CNS-ATM/EduBot
+author = Joost Ellerbroek
+author_email = j.ellerbroek@tudelft.nl
+classifiers =
+    Development Status :: 3 - Alpha
+    Intended Audience :: Developers
```
Comment (jooste) [inline setup.cfg]: Is this the best audience category for edubot?

File: setup.cfg [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    License :: OSI Approved :: Apache License Version 2.0
+    Operating System :: OS Independent
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Topic :: Software Development :: Build Tools
```
Comment (jooste) [inline setup.cfg]: Also maybe not the best category?

**Review by jooste — APPROVED** [review]
Left a few comments in setup.cfg, nothing major. We can also fix those afterwards. I'm not much of a formatting enforcer, so the recommendations in CONTRIBUTING.md are a bit heavy for me :)

**Comment by jooste** [comment]
Again a merge conflict. Too many people committing :)

**Comment by skilkis** [comment]
> 
> 
> Left a few comments in setup.cfg, nothing major. We can also fix those afterwards. I'm not much of a formatting enforcer, so the recommendations in CONTRIBUTING.md are a bit heavy for me :)

I'm working on fixing the metadata and integrating #15 now. It will be ready soon. I also toned down the best-practices in `CONTRIBUTING.md` which was left over from my own project. For me the code formatting is more of a convenience than anything, it's nice to have something take care it automatically and we can make it as lenient as we want 😃

**Comment by skilkis** [comment]
@jooste what do you think of the following metadata?

https://github.com/skilkis/EduBot/blob/48ea7a69a6137f1006568ab6af6870e5763cad08/setup.cfg#L18-L41

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+author_email = j.ellerbroek@tudelft.nl
+classifiers =
+    Development Status :: 3 - Alpha
+    Intended Audience :: Education
+    License :: OSI Approved :: GNU Affero General Public License v3
+    Operating System :: OS Independent
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Topic :: Communications :: Chat
+    Topic :: Education
```
Comment (skilkis) [inline setup.cfg:41]: @jooste What do you think of the corrected metadata? There are a list of available classifiers here: https://pypi.org/classifiers/

Comment (jooste) [reply setup.cfg:41]: Looks good!

**Pull request merged**